### DISCUSSION TITLE
useClickOutside and event propagation

### DISCUSSION THREAD

**Discussion by blcoyote** [mantinedev/mantine#5977] [Q&A]
This might be a dumb question, so bear with me.

I have a simle usecase scenaro for the useClickOutside hook.

I create a ref:
`const ref = useClickOutside(() => {
			opened && toggle();
	}, ["mouseup", "touchend"]);`
that I pass into a drawer component.

the useClickOutside function does work, closing the drawer when I click outside the component.

However the click event passes on down through the "stack" and triggers either elements on the page below the drawer, or if unfortunate the "drawer open" burger button.
Normally I would stop this behaviour with event.stopPropagation(), but being new to mantine i have no clue where to accomplish this as its just a hook passed in a ref.

the drawer state is handled like the mantine examples with `const [opened, { toggle }] = useDisclosure();`

**Comment by rtivital** [comment]
If you need to listen for events on multiple nodes, use https://mantine.dev/hooks/use-click-outside/#multiple-nodes

**Reply by blcoyote** [reply]
That doesn't solve my problem though. If I click outside the drawer to close it, I'd have to except every other interactable element on the page this way. And then just hope the user doesn't click one of them to close the drawer 

In mui for instance, when you click outside a drawer/modal/etc. You can stop the click event from travelling further by stopping its propagation. I don't see the event being even interactable in this situation.
